modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/AbstractKeyValueStorage.java

```diff
@@ -334,18 +314,13 @@ public void watchExact(byte[] key, long rev, WatchListener listener) {
     }
 
     @Override
-    public CompletableFuture<Void> readOperationsFuture(long compactionRevisionExcluded) {
-        return readOperationForCompactionTracker.collect(compactionRevisionExcluded);
-    }
-
-    @Override
-    public void registerCompactionListener(CompactionListener listener) {
-        compactionListeners.add(listener);
+    public void registerCompactionRevisionUpdateListener(CompactionRevisionUpdateListener listener) {
```

Comment:
Why don't you reuse `OnRevisionAppliedCallback`? It has a weird name, I get it, but it's a good place. There's no need for an additional list of listeners. Or am I wrong?

Reply:
fix it